src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,180 @@
+    }
+
+    @Test
+    void register_inadmissibleAge_NotOK() {
+        george.setAge(ZERO_AGE);
+        george.setLogin(anotherValidLogin);
+        george.setPassword(anotherValidPassword);
+        jane.setAge(NEGATIVE_AGE);
+        jane.setLogin(thirdValidLogin);
+        jane.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(george));
+        assertThrows(InvalidDataException.class, () -> registrationService.register(jane));
+    }
+
+    @Test
+    void register_shortLogin_NotOk() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(invalidLogin);
+        john.setPassword(validPassword);
+        george.setAge(ANOTHER_ADULT_AGE);
+        george.setLogin(shortLogin);
+        george.setPassword(anotherValidPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+        assertThrows(InvalidDataException.class, () -> registrationService.register(george));
+    }
+
+    @Test
+    void register_shortPassword_NotOk() {
+        john.setAge(ANOTHER_ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(invalidPassword);
+        george.setAge(ADULT_AGE);
+        george.setLogin(thirdValidLogin);
+        george.setPassword(shortPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+        assertThrows(InvalidDataException.class, () -> registrationService.register(george));
+    }
+
+    @Test
+    void register_emptyLogin_NotOk() {
+        john.setAge(ADULT_AGE);
+        john.setLogin("");
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void register_emptyPassword_NotOK() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword("");
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void register_nullUser_NotOk() {
+        assertThrows(InvalidDataException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_nullLogin_NotOk() {
+        john.setAge(ANOTHER_ADULT_AGE);
+        john.setLogin(null);
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void register_nullPassword_NotOk() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(null);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void register_duplicateUser_NotOk() {
+        john.setAge(ANOTHER_ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(validPassword);
+        Storage.people.add(john);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
```

Comment:
This test checks for duplicate user registration, which is good. However, make sure that the `register` method in `RegistrationServiceImpl` contains logic to actually check for duplicates. Otherwise, this test may pass without the correct implementation.